core/templates/I18nFooter.ts

```diff
@@ -50,18 +50,8 @@ angular.module('oppia').directive('i18nFooter', [
           };
           ctrl.$onInit = function() {
             ctrl.supportedSiteLanguages = SUPPORTED_SITE_LANGUAGES;
-
-            // The $timeout seems to be necessary for the dropdown
-            // to show anything at the outset, if the default language
-            // is not English.
-            $timeout(function() {
-              // $translate.use() returns undefined until the language
-              // file is fully loaded, which causes a blank field
-              // in the dropdown, hence we use $translate.proposedLanguage()
-              // as suggested in http://stackoverflow.com/a/28903658
-              ctrl.currentLanguageCode = $translate.use() ||
-                $translate.proposedLanguage();
-            }, 50);
+            ctrl.currentLanguageCode = (
+              $translate.proposedLanguage() || $translate.use());
```

Comment:
@shavavo, Wanted to confirm that you have test this in different browser at differnt internet speed? [I'll approve this once you'll confirm.]

[Also, if this is async call should we display `Languages: Loding...` while this call is not complete? [Not sure] any thoughts?]

Reply:
@DubeySandeep I've tested with throttled speed (slow 3G) and found that the selector is able to retrieve the language instantly (makes sense, because it is retrieving it locally from a token). Given that, I don't think a "Loading..." indicator is warranted.